apps/admin-portal/src/components/users/add-user.tsx

```diff
@@ -50,14 +51,86 @@ export const AddUser: React.FunctionComponent<AddUserProps> = (props: AddUserPro
 
     const [ userStoreOptions, setUserStoresList ] = useState([]);
     const [ passwordOption, setPasswordOption ] = useState(initialValues && initialValues.passwordOption);
+    const [ isUsernameValid, setIsUsernameValid ] = useState<boolean>(true);
+    const [ updatedUsername, setUpdatedUsername ] = useState<string>(initialValues?.userName);
+    const [ userStore, setUserStore ] = useState<string>("");
+    const [ randomPassword, setRandomPassword ] = useState<string>("");
+    const [ isPasswordGenerated, setIsPasswordGenerated ] = useState<boolean>(false);
 
     const { t } = useTranslation();
 
+    /**
+     * The following useEffect is triggered when a random password is generated.
+     */
+    useEffect(() => {
+        if (randomPassword && randomPassword !== "") {
+            setIsPasswordGenerated(true);
+        }
+    }, [ randomPassword ]);
+
+    /**
+     * The following useEffect is triggered when the username gets updated.
+     */
+    useEffect(() => {
+        setIsUsernameValid(false);
+        validateUsername(updatedUsername);
+    }, [ updatedUsername ]);
+
+    /**
+     * The following useEffect is triggered when the username gets updated.
+     */
+    useEffect(() => {
+        setIsUsernameValid(false);
+        validateUsername(updatedUsername);
+    }, [ userStore ]);
+
+    /**
+     * Fetch the list of available user stores.
+     */
+    useEffect(() => {
+        getUserStores();
+    }, []);
+
     const passwordOptions = [
         { label: "Invite user to set password", value: "askPw" },
         { label: "Set user password", value: "createPw" },
     ];
 
+    /**
+     * The following function validates whether the username entered by the user already exists in the
+     * user store selected by the user.
+     *
+     * @param username
+     */
+    const validateUsername = (username: string) => {
+        getUsersList(null, null, "userName eq " + username, null, userStore)
+            .then((response) => {
+                setIsUsernameValid(response?.totalResults === 0);
+            });
+    };
+
+    /**
+     * The following function handles the change of the userstore.
+     *
+     * @param values
+     */
+    const handleUserStoreChange = (values) => {
+        setUserStore(values.get("domain").toString());
+    };
+
+    /**
+     * The following function handles the change of the username.
+     *
+     * @param values
+     */
+    const handleUserNameChange = (values) => {
+        setUpdatedUsername(values?.get("userName")?.toString());
+    };
+
+    const generateRandomPassword = () => {
+        setRandomPassword(generate({ length: 10, numbers: true }));
+    };
+
```

Comment:
Function parameters and return types should be added.

Reply:
Resolved with e7e4251